From bdb58184e1a48f42a1d04506c675624e55939c8a Mon Sep 17 00:00:00 2001
From: Wright Feng <wright.feng@cypress.com>
Date: Fri, 1 Jun 2018 02:47:24 -0500
Subject: [PATCH 031/118] Revert "brcmfmac: fix CLM load error for legacy chips
 when user helper is enabled"

This reverts commit b56e808f2e14c364ea907ce183be3ed7affb5b9c.

Signed-off-by: Wright Feng <wright.feng@cypress.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
index 5eac848..a0f1eb15 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
@@ -45,8 +45,6 @@
 
 #define BRCMF_DEFAULT_TXGLOM_SIZE	32  /* max tx frames in glom chain */
 
-#define CLM_LOAD_RETRIES 1 /* # of retries to load clm_blob file */
-
 static int brcmf_sdiod_txglomsz = BRCMF_DEFAULT_TXGLOM_SIZE;
 module_param_named(txglomsz, brcmf_sdiod_txglomsz, int, 0);
 MODULE_PARM_DESC(txglomsz, "Maximum tx packet chain size [SDIO]");
@@ -180,7 +178,6 @@ static int brcmf_c_process_clm_blob(struct brcmf_if *ifp)
 	u16 dl_flag = DL_BEGIN;
 	u32 status;
 	s32 err;
-	uint retries = 0;
 
 	brcmf_dbg(TRACE, "Enter\n");
 
@@ -191,18 +188,11 @@ static int brcmf_c_process_clm_blob(struct brcmf_if *ifp)
 		return err;
 	}
 
-	do {
-		err = request_firmware(&clm, clm_name, dev);
-	} while (err == -EAGAIN && retries++ < CLM_LOAD_RETRIES);
+	err = request_firmware(&clm, clm_name, dev);
 	if (err) {
 		if (err == -ENOENT) {
 			brcmf_dbg(INFO, "continue with CLM data currently present in firmware\n");
 			return 0;
-		} else if (err == -EAGAIN) {
-			brcmf_dbg(INFO, "reached maximum retries(%d)\n",
-				  CLM_LOAD_RETRIES);
-			brcmf_dbg(INFO, "continue with CLM data in firmware\n");
-			return 0;
 		}
 		brcmf_err("request CLM blob file failed (%d)\n", err);
 		return err;
-- 
1.9.1

